sympy/physics/mechanics/jointsmethod.py

```diff
@@ -0,0 +1,132 @@
+from sympy.physics.mechanics import KanesMethod
+
+__all__ = ['JointsMethod']
+
+
+class JointsMethod(object):
+    """
+    Joints method object.
+
+    This object is used to form the equations of motion of a system created
+    using Joints.
+
+    Parameters
+    ----------
+    joints: List of Joints
+        List of all the joints created must be passed as argument.
+    root_body: Body
+        Root body w.r.t which equations of motion much be generated.
+        Is necessary to remove the  ambiguity since user may not
+        connect all bodies using joints and thus, there is not
+        way to find the root_body.
+
+    Examples
+    --------
+    >>> from sympy import Symbol
+    >>> from sympy.physics.mechanics import (Body, PinJoint, SphericalJoint, \
+                                             JointsMethod)
+    >>> root_body = Body('root_body')
+    >>> parent = Body('parent')
+    >>> child = Body('child')
+    >>> l1 = Symbol('l1')
+    >>> joints = []
+    >>> pin_joint = PinJoint('pin_joint' root_body, parent, \
+                             child_point_pos=(0, l1, 0))
+    >>> joints.append(pin_joint)
+    >>> l2 = Symbol('l2')
+    >>> spherical_joint = SphericalJoint('spherical_joint', parent, child, \
+                                         child_point_pos=(0, l2, 0))
+    >>> joints.append(spherical_joint)
+    >>> JM = JointsMethod(joints, root_body)
```

Comment:
Would it be better to do this:

```python
method = JointsMethod(root_body, joint1, joint2)
```
or
```python
method = JointsMethod(root_body, *joints)
```

I think `root_body` should always come first and I think it is more pythonic to accept any number of joints instead of an iterable of joints.

Reply:
@moorepants wouldn't it be better to allow user to pass a frame rather than root_body?

`method = JointsMethod(frame , *joints)`

Reply:
I don't know. I haven't looked at this stuff in years and can't remember what all the decisions were.

My suggestion is to take some existing example problems from the existing tests and rewrite them using this proposed method. Starting with examples will show us the issues and holes in the design.

Reply:
Yes , it seems like a good idea to me.